initializr-generator-test/src/main/java/io/spring/initializr/generator/test/buildsystem/maven/MavenBuildAssert.java

```diff
@@ -298,7 +298,7 @@ public MavenBuildAssert hasRepository(String id, String name, String url, Boolea
 				}
 				if (url != null) {
 					try {
-						new UrlAssert(repository.getUrl()).isEqualTo(new URL(url));
+						new UrlAssert(repository.getUrl()).isEqualToWithSortedQueryParameters(new URL(url));
```

Comment:
Sorry but I don't understand why we should relax the assertion here. The URL is provided as is by the code so it sounds logical to me that you'd test it's been provided as is rather than ignoring that the URL is formatted differently.

Reply:
new UrlAssert(url1).isEqualTo(url2); // Does NOT correctly compare two URLs - will NOT throw exception if two URLs differ
new UrlAssert(url1).isEqualToWithSortedQueryParameters(url2); // will throw exception is two URLs differ.

Reply:
@fg78nc thanks for the feedback but stating it won't throw an exception if the URLs differ is not telling me in what way this will happen. As I've requested in my review, can you please share the use case? At the very least we should have additional tests that cover those scenarios and make sure they are working properly then.

Reply:
There is a test already that fails because URLs are not compared correctly (exception is not thrown)
https://github.com/spring-io/initializr/blob/eed9c3876f4114fd818bfb989936e00b0f62c6f2/initializr-generator-test/src/test/java/io/spring/initializr/generator/test/buildsystem/maven/MavenBuildAssertTests.java#L306